### ISSUE TITLE
"the trait bound `actix_http::header::ExtendedValue: std::cmp::Eq` is not satisfied" when use zip crate

### ISSUE THREAD

**Issue by gituseridk** [actix/actix-web#2872]
## Expected Behavior
_actix-web_ must be compiled when using _zip_ crate in dependencies

## Current Behavior
Failed to compile _actix-web_ when use _zip_ crate in dependencies with this error:
```
Compiling actix-web v4.2.0
error[E0277]: the trait bound `actix_http::header::ExtendedValue: std::cmp::Eq` is not satisfied
  --> /Users/user/.cargo/registry/src/github.com-1ecc6299db9ec823/actix-web-4.2.0/src/http/header/content_disposition.rs:99:17
   |
82 | #[derive(Clone, Debug, PartialEq, Eq)]
   |                                   -- in this derive macro expansion
...
99 |     FilenameExt(ExtendedValue),
   |                 ^^^^^^^^^^^^^ the trait `std::cmp::Eq` is not implemented for `actix_http::header::ExtendedValue`
   |
note: required by a bound in `AssertParamIsEq`
   = note: this error originates in the derive macro `Eq` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `actix-web` due to previous error
```

## Steps to Reproduce
1. `cargo init`
2. add this to Cargo.toml:
```toml
[dependencies]
actix-web = "4"
zip = "0.6"
```
3. `cargo run`

## Environment
- MacBook Air (M1, 2020).
- Macos 12.5.1 (21G83).
- Rust Version: `rustc 1.63.0`
- Actix Web Version: `v4.2.0`

**Comment by gituseridk** [comment]
For now I fix this by downgrading _zip_ version to "0.5"  

```toml
# Cargo.toml
[dependencies]
actix-web = "4"
zip = "0.5.0"
```

**Comment by JohnTitor** [comment]
Duplicate of https://github.com/actix/actix-web/issues/2869, the issue was fixed by v4.2.1.
